common/persistence/sql/sqlHistoryManager.go

```diff
@@ -104,6 +104,9 @@ func (m *sqlHistoryManager) GetWorkflowExecutionHistory(request *p.InternalGetWo
 
 	// TODO: Ensure that no last empty page is requested
 	if err == sql.ErrNoRows || (err == nil && len(rows) == 0) {
+		if len(request.NextPageToken) != 0 {
```

Comment:
Are you able to just always return an empty response regardless of NextPageToken, this will get handled correctly at upper layer?

Reply:
I'd like to match the existing behavior in Cassandra V1 here.

Reply:
If there is no big change, I think it's better to let both V1/V2 return this error in upper layer instead of lower level. As we always try to keep our lower level code simple so that adding new implementation (like Spanner/DynamoDB/etc) would be much easier.